src/util/system.cpp

```diff
@@ -945,6 +945,14 @@ bool ArgsManager::ReadConfigStream(std::istream& stream, const std::string& file
     for (const std::pair<std::string, std::string>& option : options) {
         KeyInfo key = InterpretKey(option.first);
         std::optional<unsigned int> flags = GetArgFlags('-' + key.name);
+        if (key.name == "conf") {
+            error = "conf cannot be set in the configuration file; if you want to include additional configuration files, use includeconf";
+            return false;
+        }
+        if (key.name == "reindex") {
+            error = "reindex cannot be set in the configuratoin file; use -reindex=1 from the commandline";
+            return false;
+        }
```

Comment:
quick thing:
 would be better to decouple this into a different function `IsConfKeySupported` (or something similar) to keep `ArgsManager::ReadConfigStream` as general as possible and not pollute it with the inclusion of every single not supported option (might be more in the future).

Reply:
great suggestion; added!